src/providers/RequestQueue.js

```diff
@@ -0,0 +1,51 @@
+import React, { useCallback, useContext, useState } from 'react'
+import PropTypes from 'prop-types'
+
+const RequestQueueContext = React.createContext()
+
+function RequestQueueProvider({ children }) {
+  const [requestQueue, setRequestQueue] = useState([])
+
+  const addRequest = useCallback(
+    request => {
+      const newRequests = [...requestQueue, request]
+      return setRequestQueue(newRequests)
+    },
+    [requestQueue]
+  )
+
+  const addRequests = useCallback(
+    requests => {
+      const newRequests = [...requestQueue, ...requests]
```

Comment:
Note that with `array.concat()` you effectively can do both using the same API, so we could just have `addRequests()`.